File: Codes/3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i.go
/*
 * @Author: LetMeFly
 * @Date: 2024-10-18 23:06:33
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-10-18 23:08:12
 */
package main

func minOperations(nums []int) int {
    ans := 0
    for i := 0; i < len(nums) - 2; i++ {
        if nums[i] == 0 {
            ans++
            nums[i + 1] ^= 1
            nums[i + 2] ^= 1
        }
    }
    if nums[len(nums) - 1] & nums[len(nums) - 2] == 1 {
        return ans
    } else {
        return -1
    }
}

File: Codes/3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i.py
'''
Author: LetMeFly
Date: 2024-10-18 23:02:03
LastEditors: LetMeFly
LastEditTime: 2024-10-18 23:03:25
'''
from typing import List

class Solution:
    def minOperations(self, nums: List[int]) -> int:
        ans = 0
        for i in range(len(nums) - 2):
            if not nums[i]:
                ans += 1
                nums[i + 1] ^= 1
                nums[i + 2] ^= 1
        return ans if nums[-1] & nums[-2] else -1
